manifests/freeipa_user_provider.pp

```diff
@@ -0,0 +1,67 @@
+#
+# @summary setup IPA as an LDAP user provider for Keycloak
+#
+# @example Add FreeIPA as a user provider
+#   keycloak::freeipa_user_provider { 'ipa.example.org':
+#     ensure          => 'present',
+#     realm           => 'EXAMPLE.ORG',
+#     bind_dn         => 'uid=ldapproxy,cn=sysaccounts,cn=etc,dc=example,dc=org',
+#     bind_credential => 'secret',
+#     users_dn        => 'cn=users,cn=accounts,dc=example,dc=org',
+#     priority        => 10,
+#   }
+#
+# @param ensure
+#   LDAP user provider status
+# @param ipa_host
+#   Hostname of the FreeIPA server (e.g. ipa.example.org)
+# @param realm
+#   Keycloak realm
+# @param bind_dn
+#   LDAP bind dn
+# @param bind_credential
+#   LDAP bind password
+# @param users_dn
+#   The DN for user search
+# @param priority
+#   Priority for this user provider
+#
+define keycloak::freeipa_user_provider
+(
+  String                    $realm,
+  String                    $bind_dn,
+  String                    $bind_credential,
+  String                    $users_dn,
+  Enum['present', 'absent'] $ensure = 'present',
+  Stdlib::Host              $ipa_host = $title,
+  Integer                   $priority = 10,
+  Boolean                   $ldaps = false,
```

Comment:
Also looks like tests are failing because `ldaps` is not documented at top of file.

Reply:
Fixed the documentation issue. Hopefully Travis becomes happy. FreeIPA uses the LDAP protocol and port 389 by default. I'm not sure why, because there's no shortage of certificates in FreeIPA, it being a CA among other things.